### ISSUE TITLE
‘getUnalnCtg’ function has some problem!

### ISSUE THREAD

**Issue by Bobo22331** [SJTU-CGM/HUPAN#3]
When I run the HUPAN like this
```
hupan getUnalnCtg /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/ /raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result /raid8/cuixb/data/Bna/60x_assemble/hupan_result/Unalign_result
```
it gives the `Warning: cannot find TH001 in /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/TH001/ `,
but my contig file is in it. Can you help to solve this? Thank you.

**Comment by zhqduan** [comment]
In this step, you should provide the contig file of TH001 in  /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/TH001/ and also the **quast result** of TH001 in "/raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result/TH001".
Only all the above two directory are available, this step works.

**Comment by Bobo22331** [comment]
Yeah, the contig file is in the "/raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/TH001/" directory, but it also gives the warning: "Warning: cannot find TH001 in /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/TH001 ", my command is：
"hupan getUnalnCtg -p .contig /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/TH001 /raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result/data/ /raid8/cuixb/data/Bna/60x_assemble/hupan_result/Unalign_result", 
I think maybe some problem  in the HUPAN/lib/HUPANunalnCtg.pm line 89-98.

**Comment by zhqduan** [comment]
The command should be "hupan getUnalnCtg -p .contig /raid8/cuixb/data/Bna/60x_assemble/hupan_result/candidate/data/ /raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result/data/ /raid8/cuixb/data/Bna/60x_assemble/hupan_result/Unalign_result" .
**Did you have the quast result of TH001 as "/raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result/data/TH001".** 
I checked the codes in the HUPAN/lib/HUPANunalnCtg.pm line 89-98, it seems like okey.

**Comment by Bobo22331** [comment]
yeah, it's ok now, but getUnalnCtg.pl also give warning with: "Warnings: cannot find fully unaligned file(*.fully.contig) from sample: TH001". So, what's the fully.contig file, it's the file in the candidate/data ?

**Comment by Bobo22331** [comment]
The problem still exists, can you check the source code of getUnalnCtg.pl, there maybe some wrong in it . Thank you!

**Comment by Bobo22331** [comment]
The command is: `hupan getUnalnCtg -p .contig candidate/data/ quast_result/data/ unalign_result`
The output is: 
```Process sample TH001_BDSW190036040-1a ...

Usage: /raid8/cuixb/tools/biosoft/hupan/bin/getUnalnCtg.pl [options] <contig_file> <unaligned_info_file> <coords_file> <output_prefix> <prefix>

/raid8/cuixb/tools/biosoft/hupan/bin/getUnalnCtg.pl is used to collect both fully unaligned contigs and partially unaligned contigs, and
collect the coords of the partially unaligned contigs.

Necessary input description:

  contig_file             <string>    Assembled contig fasta file.

  unaligned_info_file     <string>    File including the info of unaligned contigs.

  coords_file             <string>    File including the coords of contigs.

  output_prefix           <string>    The prefix of outfiles.

  prefix                  <string>    A prefix added to the name of each contig. This is used
                                      to mark which sample the contigs are from.

Options:
     -h                               Print this usage page.

Warnings: cannot find fully unaligned file(*.fully.contig) from sample: TH001_BDSW190036040-1a
```

The file in the candidate/data/ is: `TH001_BDSW190036040-1a/TH001_BDSW190036040-1a.candidate.unaligned.contig`
The file in the quast_result/data/ are:
![](https://i.loli.net/2019/11/14/QV7qYELoNnj8Xva.jpg)
![](https://i.loli.net/2019/11/14/8bkCgdphWH6lG1X.jpg)

**Comment by zhqduan** [comment]
In the HUPAN, the tool 'getUnalnCtg' extracts the unaligned contigs from QUAST result. And two different types of unaligned contings are considered: the fully unaligned contigs and the partially unaligned contigs. Please see http://quast.bioinf.spbau.ru/manual.html for the definition of fully unaligned contains and partially unaligned contigs. I doubt that the QUAST results put the wrong directory. You should name the QUAST result of TH001 as "TH001" in the /raid8/cuixb/data/Bna/60x_assemble/hupan_result/quast_result/data/.

**Comment by Bobo22331** [comment]
Ok, thank you!

**Comment by zhqduan** [comment]
I am sorry that there is a bug in the getUnalnCtg.pl and it is okey now. Please re-download. Thank you very much.

**Comment by Bobo22331** [comment]
Yeah, thank you for correct the bug, I have tried many many times but it didn't work.
